tests/zero-downtime-scaling/results/1.0.0/1.0.0.md

```diff
@@ -0,0 +1,584 @@
+    Latency     4.25ms   49.55ms   1.06s    99.35%
+    Req/Sec     1.45k   130.50     1.76k    75.07%
+  Latency Distribution
+     50%  660.00us
+     75%  735.00us
+     90%  831.00us
+     99%    2.32ms
+  731286 requests in 5.00m, 256.43MB read
+  Socket errors: connect 0, read 18, write 0, timeout 1
+Requests/sec:   2437.02
+Transfer/sec:      0.85MB
+```
+
+HTTPS graph:
+
+![gradual-up-https.png](25-node/gradual-up-https.png)
+
+HTTPS failure rate: .31%
+
+Observations:
+
+- After a minute or so, the oldest Pod switched from Ready -> Not Ready and took ~ 1 min to recover. The following event
+  was recorded on the Pod:
+
+  ```text
+  Readiness probe failed: Get "http://10.4.0.17:8081/readyz": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
+  ````
+
+Logs:
+
+- NGF error logs: none
+- NGINX error logs:
+
+  Handful of these errors and warnings:
+
+  ```text
+  2023/10/24 17:31:43 [warn] 176#176: *358453 upstream server temporarily disabled while connecting to upstream, client: 10.X.X.X:, server: cafe.example.com, request: "GET /tea HTTP/1.1", upstream: "http://10.X.X.X:8080/tea", host: "cafe.example.com"
+  2023/10/24 17:31:43 [error] 176#176: *358453 upstream timed out (110: Operation timed out) while connecting to upstream, client: 10.X.X.X, server: cafe.example.com, request: "GET /tea HTTP/1.1", upstream: "http://10.X.X.X:8080/tea", host: "cafe.example.com"
+  ```
+
+- NGINX access logs:
+  - 1.5 million 200s
+
+### Scale Down Gradually
+
+HTTP wrk output:
+
+```text
+Running 20m test @ http://cafe.example.com/coffee
+  2 threads and 2 connections
+    Thread Stats   Avg      Stdev     Max   +/- Stdev
+    Latency    15.01ms   98.51ms   1.06s    97.53%
+    Req/Sec     1.54k   175.82     1.85k    86.64%
+  Latency Distribution
+     50%  629.00us
+     75%  701.00us
+     90%  800.00us
+     99%  662.41ms
+  2548003 requests in 17.44m, 0.89GB read
+  Socket errors: connect 0, read 15, write 0, timeout 50
+Requests/sec:   2434.36
+Transfer/sec:      0.87MB
+```
+
+HTTP graph:
+
+![gradual-down-http.png](25-node/gradual-down-http.png)
+
+HTTP failure rate .26%
+
+HTTPS wrk output:
+
+```text
+Running 20m test @ https://cafe.example.com/tea
+  2 threads and 2 connections
+    Thread Stats   Avg      Stdev     Max   +/- Stdev
+    Latency    22.15ms  119.12ms   1.07s    96.38%
+    Req/Sec     1.46k   178.46     1.76k    90.82%
+  Latency Distribution
+     50%  658.00us
+     75%  736.00us
+     90%    0.86ms
+     99%  780.01ms
+  2427687 requests in 17.44m, 851.30MB read
+  Socket errors: connect 0, read 15, write 0, timeout 53
+Requests/sec:   2319.78
+Transfer/sec:    832.98KB
+```
+
+HTTPs graph:
+
+![gradual-down-https.png](25-node/gradual-down-https.png)
+
+HTTPS failure rate .27%
+
+Logs:
+
+- NGF logs: none
+- NGINX error logs: none
+  - NGINX access logs:
+    - ~ 5 million 200s
+    - non-200s:
+
+      ```text
+      10.168.0.14 - - [24/Oct/2023:17:58:47 +0000] "GET /coffee HTTP/1.1" 499 0 "-" "-"
```

Comment:
those are probably related to wrk and curl stopping sending requests -- https://github.com/nginxinc/nginx-gateway-fabric/blob/main/tests/zero-downtime-upgrades/results/1.0.0/1.0.0.md#new-pods

Reply:
yep, makes sense. I'll add a note here.